nemo/lightning/pytorch/accelerate/transformer_engine.py

```diff
@@ -0,0 +1,90 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from types import MethodType
+
+import torch
+from transformer_engine import pytorch as te
```

Comment:
Searched in NeMo/requirements and did not find transformer_engine, so I'm thinking importing here might break other paths (e.g. ASR) that don't have TE installed; can we use safe_import instead?
```
from nemo.utils.import_utils import safe_import_from
te, HAVE_TE = safe_import_from("transformer_engine", "pytorch")
```
And then later add an assert statement for HAVE_TE wherever it's used?

Reply:
If we install mcore, TE should be included actually. Let me check this.

Reply:
Ok, added the safe import.